src/main/java/forestry/climatology/PreviewHandlerClient.java

```diff
@@ -0,0 +1,168 @@
+
+import javax.annotation.Nullable;
+import java.util.HashSet;
+import java.util.Set;
+
+import net.minecraft.block.Block;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.RenderGlobal;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import net.minecraftforge.client.event.RenderWorldLastEvent;
+import net.minecraftforge.event.world.WorldEvent;
+
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.gameevent.TickEvent;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import forestry.api.climate.ClimateCapabilities;
+import forestry.api.climate.IClimateTransformer;
+import forestry.climatology.items.ItemHabitatScreen;
+import forestry.core.tiles.TileUtil;
+import forestry.core.utils.TickHelper;
+
+@SideOnly(Side.CLIENT)
+public class PreviewHandlerClient {
+
+	public static final PreviewHandlerClient INSTANCE = new PreviewHandlerClient();
+
+	private static final AxisAlignedBB BOUNDING_BOX = Block.FULL_BLOCK_AABB.shrink(0.125F);
+
+	private TickHelper tickHelper = new TickHelper();
+	private final Set<BlockPos> previewPositions = new HashSet<>();
+	@Nullable
+	private BlockPos previewOrigin = null;
+	private int range = -1;
+	private boolean circular;
+
+
+	private PreviewHandlerClient() {
+	}
+
+	public void setPreview(BlockPos centerPosition, int range, boolean circular) {
+		previewPositions.clear();
+		BlockPos center = new BlockPos(0, 0, 0);
+		for (int x = -range; x <= range; x++) {
+			for (int y = -range; y <= range; y++) {
+				BlockPos position = centerPosition.add(x, 0, y);
+				boolean valid;
+				if (circular) {
+					double distance = Math.round(center.getDistance(x, 0, y));
+					valid = distance <= range && distance > (range - 1);
+				} else {
+					valid = !(!(x == -range || x == range) && !(y == -range || y == range));
+				}
+				if (valid) {
+					previewPositions.add(position);
+				}
+			}
+		}
+		previewOrigin = centerPosition;
+		this.range = range;
+		this.circular = circular;
+	}
+
+	public boolean isPreviewOrigin(BlockPos pos){
+		return previewOrigin != null && pos.equals(previewOrigin);
+	}
+
+	@Nullable
+	public BlockPos getPreviewOrigin() {
+		return previewOrigin;
+	}
+
+	public void updatePreview(BlockPos centerPosition, int range, boolean circular) {
+		if (centerPosition == previewOrigin || range != this.range || circular != this.circular) {
+			setPreview(centerPosition, range, circular);
+		}
+	}
+
+	public void clearPreview() {
+		previewOrigin = null;
+		circular = false;
+		range = -1;
+		previewPositions.clear();
+	}
+
+	@SubscribeEvent
+	public void onWorldRenderLast(RenderWorldLastEvent event) {
```

Comment:
This event is called so much that for performance you may want to make a special class to handle it. That class can subscribe/unsubscribe to the event if we have previewPositions to show.